src/cascade/input_data/db/csmr.py

```diff
@@ -0,0 +1,136 @@
+"""Upload csmr data to t3 table so EpiViz can access it for plotting."""
+
+import logging
+
+try:
+    from db_queries import get_outputs
+except ImportError:
+
+    class DummyGetOutputs:
+        def __getattr__(self, name):
+            raise ImportError(f"Required package db_queries not found")
+
+    get_outputs = DummyGetOutputs()
+
+
+from cascade.core.db import cursor
+
+
+CODELOG = logging.getLogger(__name__)
+
+
+def _csmr_in_t3(execution_context):
+    """Checks if data for the current model_version_id already exists in tier 3.
+    """
+
+    model_version_id = execution_context.parameters.model_version_id
+
+    query = """
+    SELECT exists(
+             SELECT * FROM epi.t3_model_version_csmr
+             WHERE model_version_id = %(model_version_id)s
+    )
+    """
+    with cursor(execution_context) as c:
+        c.execute(query, args={"model_version_id": model_version_id})
+        exists = c.fetchone()[0]
+
+    return exists == 1
+
+
+def _get_csmr_data(execution_context):
+
+    model_version_id = execution_context.parameters.model_version_id
+    add_csmr_cause = execution_context.parameters.add_csmr_cause
+    parent_loc = execution_context.parameters.drill
+
+    csmr_measure_id_deaths = 1  # measure type = deaths (1)
+    csmr_metric_id_rate = 3  # metric type = per capita rate
+    csmr_gbd_round_id = 5
+
+    csmr_keep_cols = ["year_id", "location_id", "sex_id", "age_group_id",
+                      "val", "lower", "upper"]
+
+    csmr = get_outputs(
+        topic="cause",
+        cause_id=add_csmr_cause,
+        location_id=parent_loc,
+        metric_id=csmr_metric_id_rate,
+        year_id="all",
+        age_group_id="most_detailed",
+        measure_id=csmr_measure_id_deaths,
+        sex_id="all",
+        gbd_round_id=csmr_gbd_round_id
+        )[csmr_keep_cols]
+
+    csmr["model_version_id"] = model_version_id
+
+    csmr_columns = ["model_version_id"] + csmr_keep_cols
+    csmr = csmr[csmr_columns]
+
+    csmr.rename(columns={"val": "mean"}, inplace=True)
+
+    return csmr
+
+
+def _upload_csmr_data_to_tier_3(cursor, model_version_id, csmr_data):
+    """Uploads csmr data to tier 3 attached to the current model_version_id.
+    """
+
+    insert_query = f"""
+        INSERT INTO epi.t3_model_version_csmr (
+            model_version_id,
+            year_id,
+            location_id,
+            sex_id,
+            age_group_id,
+            mean,
+            lower,
+            upper
+        ) VALUES (
+            {model_version_id}, {", ".join(["%s"]*32)}
```

Comment:
There are not 32 parameters here. Make sure you try this out on the actual database to flush problems like this. I know I keep harping on how we can't have database access in the unit tests but we do need to actually test stuff against the database. Soon we'll have integration and/or smoke tests that will automate that but for now make sure you're doing it by hand.

Reply:
I was looking at bundle.py's:  
def _upload_bundle_data_to_tier_3(cursor, model_version_id, bundle_data):

in which I see this:
cursor.executemany(insert_query, bundle_data.values())

Is bundle_data df?  I thought it was.  I must be missing something.

In a python terminal window, I created a df called x and tried to call values().
>>> type(x)
<class 'pandas.core.frame.DataFrame'>
>>> x.values()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: 'numpy.ndarray' object is not callable

Reply:
Oh, dear. I have failed to do exactly the thing I was telling you to do. The problem is that the code I wrote doesn't work. It should be `.values` not `.values()`